src/controller/CHIPDeviceController.cpp

```diff
@@ -317,6 +327,41 @@ CHIP_ERROR DeviceController::InitControllerNOCChain(const ControllerInitParams &
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DeviceController::UpdateControllerNOCChain(const ByteSpan & noc, const ByteSpan & icac, bool enableServerInteractions,
+                                                      Crypto::P256Keypair * operationalKeypair)
```

Comment:
Is changing enableServerInteractions part of updating the NOC chain?  Seems a bit odd...

Reply:
Add a variable to record the origin server interaction state.